src/components/mdx-code-block/MdxCodeBlock.tsx

```diff
@@ -0,0 +1,186 @@
+import { ReactElement, useRef } from 'react';
+
+import { Badge } from 'src/components/badge/Badge';
+import { Button } from 'src/components/button/Button';
+import { HStack } from 'src/components/stack/HStack';
+import { CopyIcon } from 'src/icons/CopyIcon';
+import { theme } from 'src/styles/constants/theme';
+import styled from 'styled-components';
+
+import { useCopyText } from '../../utils/useCopyText';
+
+const Wrapper = styled.div`
+  width: 100%;
+  background-color: ${theme.gray1100};
+  border-radius: 12px;
+  color: ${theme.gray0};
+  margin-bottom: 16px;
+`;
+
+const CodeBlockHeader = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  background-color: ${theme.gray1000};
+  padding: 8px 16px;
+  border-top-left-radius: 12px;
+  border-top-right-radius: 12px;
+
+  p {
+    margin: 0;
+    font-size: 12px;
+    font-weight: 600;
+    text-transform: uppercase;
+  }
+`;
+
+const CodeBlockBody = styled.div`
+  padding: 16px;
+  display: flex;
+  max-height: 260px;
+  overflow: auto;
+  height: fit-content;
+
+  pre {
+    white-space: pre;
+    word-break: break-word;
+    border-radius: 0;
+    padding: 0;
+    background: transparent;
+
+    .hljs {
+      color: ${theme.gray100} !important;
+      background: transparent !important;
```

Comment:
Hmn, is this class `.hljs` using inline css? If we couldn't apply styling unless using `!important`. Would you add the commend here describe why we use !important here right here?

Reply:
The `.hljs` class is for the `highlight-js` library. I'm assuming we will be using the same library for our next MDX project. I had to use `!important` to override the libraries styles. I'm assuming it would need `!important` for the next project as well, but not positive 🤔 What do you think?

Reply:
Does the library use inline css for styling or just add a bunch of className to the consuming component?